swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else: 
+            return False
+        
+    def get_by_id(self, item_id):
+
+        for item in self.inventory:
+            if item.id == item_id:
+                return item 
+
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item) 
+            self.inventory.append(their_item)
+            other_vendor.inventory.remove(their_item)
+            other_vendor.inventory.append(my_item)
+            return True
+        
+        return False
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        elif self.inventory and other_vendor.inventory:
+            self.inventory.append(other_vendor.inventory[0])
+            other_vendor.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0]) 
+            other_vendor.inventory.remove(other_vendor.inventory[0])
```

Comment:
👀 Providing a custom implementation based on knowing exactly which item positions to swap gives us enough information that we could perform the swap in constant rather than linear time. As long as we know that we _have_ the items to swap (i.e., the inventory of neither vendor is empty), we can exchange positions of the two items directly.

The approach here has that effect, but rather than a direct exchange, which would take constant time, this implementation uses `remove` and `append`, taking linear time. The result is the same, but the implementation is inefficient.

How can we avoid the linear operations to accomplish this swap in constant time?